Tests/FoundationTests/URLSessionExtensionsTests.swift

```diff
@@ -0,0 +1,41 @@
+// URLSessionExtensionsTests.swift - Copyright 2023 SwifterSwift
+
+@testable import SwifterSwift
+import XCTest
+
+#if canImport(Foundation)
+import Foundation
+
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
+final class URLSessionExtensionsTests: XCTestCase {
+    func testDataSync() {
+        let gemfileContent = "source 'https://rubygems.org'\n\n" +
+            "gem 'xcpretty'\n" +
+            "gem 'danger'\n" +
+            "gem 'danger-swiftlint'\n" +
+            "gem 'danger-xcode_summary'\n" +
+            "gem 'xcpretty-json-formatter'\n\n" +
+            "gem 'fastlane'\n" +
+            "gem 'cocoapods'"
+        let url = URL(string: "https://raw.githubusercontent.com/SwifterSwift/SwifterSwift/5.3.0/Gemfile")!
+        var data: Data!
+        var response: URLResponse!
+        XCTAssertNoThrow((data, response) = try URLSession.shared.dataSync(with: URLRequest(url: url)))
+        let httpResponse = response as? HTTPURLResponse
+        let content = String(data: data, encoding: .utf8)
+        XCTAssertEqual(content, gemfileContent)
+        XCTAssertNotNil(httpResponse)
+        XCTAssertEqual(httpResponse!.statusCode, 200)
+        XCTAssertEqual(httpResponse!.url, url)
+    }
+    
+    func testDataSyncError() {
+        let url = URL(string: "https://something.cm/something")!
```

Comment:
Make it something more random, in case this one day becomes a real URL :)

Reply:
Hello @guykogus
Please check this function again.